### PR TITLE
AR-19993 replace deprecating constructor

### PR DESCRIPTION
**Pull request by morisamashinka**
https://yieldmetrics.atlassian.net/browse/AR-19993

### CODE REVIEW

**Comment by morisamashinka** [comment]
image
All test passed

File: Src/MongoLinqPlusPlus/Extensions.cs [hunk 1]
```diff
@@ -74,5 +75,29 @@ public static ExpressionType Flip(this ExpressionType e)
 
             throw new NotSupportedException($"Can't (or haven't implemented) flipping ExpressionType {e}");
         }
+        /// <summary>
+        /// In replace of deprecated constructor with manually assigning pid, machine, increment
+        /// </summary>
+        /// <param name="timestamp">creation time</param>
+        /// <param name="manuallyAssignRandomBytesHexString">random bytes</param>
+        /// <returns></returns>
+        public static ObjectId GenerateNewIdWithAssigningRandomBytes(DateTime timestamp, string manuallyAssignRandomBytesHexString)
```
Comment (sdcarpenter) [inline Src/MongoLinqPlusPlus/Extensions.cs:84]: What if we did something like this (Linqpad Script):
```
void Main()
{
	var date = new DateTime(2023, 1, 1);
	
	MakeRandoDateObjectId(date, new byte[] { 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0 }).Dump();
	new ObjectId(date, 0, 0, 0).Dump();

	MakeRandoDateObjectId(date, new byte[] { 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF }).Dump();
	new ObjectId(date, 16777215, -1, 16777215).Dump();

}

// You can define other methods, fields, classes and namespaces here
private static ObjectId MakeRandoDateObjectId(DateTime date, byte[] random)
{
	var newObjId = ObjectId.GenerateNewId(date).ToByteArray();
	Array.Copy(random, 0, newObjId, newObjId.Length - 8, 8);
	return new ObjectId(newObjId);
}
```

**Review by sdcarpenter — CHANGES_REQUESTED** [review]

File: Src/MongoLinqPlusPlus/Pipeline.cs [hunk 2]
```diff
@@ -531,9 +531,13 @@ private IMongoQuery BuildMongoWhereExpressionAsQuery(Expression expression, bool
 
                     // Pull the DateTime we're comparing our ObjectId.CreationDate to
                     var comparisonValue = (DateTime) constExp.Value;
+                    //var comparisonValueAsObjectIdMin = new ObjectId(comparisonValue, 0, 0, 0);
```
Comment (sdcarpenter) [inline Src/MongoLinqPlusPlus/Pipeline.cs:534]: Tiny nit, lets delete these commented out lines

**Review by sdcarpenter — APPROVED** [review]
Approved with a tiny nit

**Pull request merged**